gateway-service/src/main/java/org/java/mentorship/gateway/exception/domain/GatewayException.java

```diff
@@ -2,17 +2,20 @@
 
 import lombok.Data;
 import lombok.EqualsAndHashCode;
+import org.java.mentorship.contracts.common.dto.ErrorResponse;
 import org.springframework.http.HttpStatus;
 
 @EqualsAndHashCode(callSuper = true)
 @Data
 public class GatewayException extends RuntimeException {
 
     private final HttpStatus statusCode;
+    private final ErrorResponse errorResponse;
 
-    public GatewayException(String message, HttpStatus statusCode) {
-        super(message);
+    public GatewayException(ErrorResponse errorResponse, HttpStatus statusCode) {
```

Comment:
Generally, it's a good idea to keep the source Exception/Throwable
It might help us with debugging since we'll have the stack trace.

Thus, you should add another parameter to this constructor and do "super(errorResponse.getError(), sourceException)

Reply:
I don't use this class anywhere anymore, but I will add the constructor